crypto/keyring/keyring.go

```diff
@@ -480,31 +497,29 @@ func (ks keystore) List() ([]Info, error) {
 
 	sort.Strings(keys)
 	for _, key := range keys {
-		if strings.HasSuffix(key, infoSuffix) {
-			rawInfo, err := ks.db.Get(key)
-			if err != nil {
-				return nil, err
-			}
+		if !strings.HasSuffix(key, infoSuffix) {
+			continue
+		}
 
-			if len(rawInfo.Data) == 0 {
-				return nil, sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, key)
-			}
-			// TODO ks.cdc which method to use?
-			var ke KeyringEntry
-			err = ks.cdc.Unmarshal(rawInfo.Data, &ke)
-		//	info, err := protoUnmarshalInfo(rawInfo.Data, ks.cdc)
-			if err != nil {
-				return nil, err
-			}
+		item, err := ks.db.Get(key)
+		if err != nil {
+			return nil, err
+		}
 
-			res = append(res, ke)
+		if len(item.Data) == 0 {
+			return nil, sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, key)
 		}
+		var ke KeyringEntry
+		if err := ks.cdc.Unmarshal(item.Data, &ke); err != nil {
```

Comment:
does `ks.cdc.Unmarshal` make sense for you?

Reply:
Yes. Do you have something else in mind?